src/tests/ftest/control/dmg_system_cleanup.py

```diff
@@ -0,0 +1,98 @@
+    (SRS-326) Pool Management - automatic pool handle revocation
+
+    Test step:
+    1. Create 2 pools.
+    2. Create a container in each pool.
+    3. Write to each container to ensure connections are working.
+    4. Call dmg system cleanup host.
+    5. Write to each container again to ensure they fail
+    6. Check that cleaned up handles match expected counts.
+
+    :avocado: recursive
+    """
+
+    def test_dmg_system_cleanup_one_host(self):
+        """
+        JIRA ID: DAOS-6471
+
+        Test Description: Test dmg system cleanup.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small,dmg
+        :avocado: tags=control,dmg_system_cleanup
+        """
+        # Create 2 pools and create a container in each pool.
+        self.pool = []
+        self.container = []
+        for _ in range(2):
+            self.pool.append(self.get_pool())
+            self.container.append(self.get_container(self.pool[-1]))
+
+        # Create 5 more connections to each pool
+        pool_handles = []
+        for pool in self.pool:
+            for _ in range(5):
+                handle = self.get_pool(create=False, connect=False)
+                handle.pool = DaosPool(self.context)
+                handle.uuid = pool.uuid
+                handle.connect(2)
+                pool_handles.append(handle)
+
+        # Check to make sure we can access the pool
+        try:
+            for i in range(2):
+                self.container[i].get_params(self)
+                self.container[i].write_objects()
+        except (DaosApiError, DaosTestError) as error:
+            self.fail("Unable to write container #{}: {}".format(i, error))
+
+        # Call dmg system cleanup on the host and create cleaned pool list.
+        dmg_cmd = self.get_dmg_command()
+        result = dmg_cmd.system_cleanup(self.hostlist_servers[0], verbose=True)
+
+        # Build list of pools and how many handles were cleaned (should be 6 each)
+        actual_handle_counts = dict()
+        for res in result["response"]["results"]:
+            if res["status"] == 0:
+                actual_handle_counts[res["pool_id"].lower()] = res["count"]
+        # Attempt to access the pool again (should fail)
+        for i in range(2):
+            try:
+                self.container[i].write_objects()
+            except (DaosApiError, DaosTestError, TestFail) as error:
+                self.log.info("Unable to write container #%d: as expected %s", i, error)
+            else:
+                self.fail("Wrote to container #{} when it should have failed: {}".format(i, error))
+
+        # Build a list of pool IDs and counts (6) to compare against
+        # our cleanup results.
+        expected_handle_count = dict()
+        for pool in self.pool:
+            expected_handle_count[pool.uuid.lower()] = 6
+
+        # Clear pool and container list to avoid trying to destroy them.
+        self.pool = []
+        self.container = []
```

Comment:
We don't want to leave pools on the system after the test. Maybe I'm misunderstanding something -- Do we not need to clean these up? Maybe we need to get new handles to do so.

Reply:
The pools will be destroyed when storage is formatted when starting the next test. The issue is that those objects are defunct at this point because all of their handles are gotten rid of.

Reply:
I'm not sure this assertion is true. The servers are not reformatted before each test, and haven't been for a long time. That was changed in an effort to improve CI throughput. I'm going to add some more reviewers for comment on the test code.

Reply:
Using `self.pool` and `self.container` as lists with the `self.get_pool()` and `self.get_container()` to populate them, respectively, will result in `TestWithServers.tearDown()` calling `self.destroy_pools(self.pool)` and `self.destroy_containers(self.container)` to remove any pools and containers defined in those lists.  These methods are called even when the servers are left continually running.  I have also recently added code to run `dmg pool list` and destroy any pools found at the end of `tearDown()` and in `setUp()` when running servers continuously.  If there are errors found stopping any pools found in the dmg pool list output the servers are stopped to ensure the next test variant/method will format the storage and restart the servers.  In short, the test implements this correctly.

Reply:
What I'm less sure about is having multiple open pool connections and not having a means to ensure they are closed before attempting to destroy the pools.  Does using the force flag on the dmg pool destroy handle this?